src/verifications/verifications.ts

```diff
@@ -1515,6 +1545,31 @@ export class Verifications extends Logger {
     return true;
   }
 
+  /**
+   * checks if a address is an identity or not
+   *
+   * @param {string} identityAddress The identity address to fetch the owner for.
+   * @returns {Promise<boolean>} identity or not.
+   */
+  public async isIdentity(address: string): Promise<boolean> {
+    try {
+      if (this.cachedAccountIdentities[address] || address.length === 66) {
+        return true;
+      }
+      const nonce = await this.options.executor.executeContractCall(
+        this.options.contractLoader.loadContract('KeyHolder', address),
+        'getExecutionNonce',
+      );
+      if (nonce !== null) {
+        this.cachedAccountIdentities[address] = true;
+        return true;
+      }
+      return false;
+    } catch (e) {
+      return false;
```

Comment:
Could this lead to trouble in case the executor or the contractLoader throw some kind of unrelated error?

Reply:
would also be possible to remove this. old web3 throwed an error when calling a function which not exists. currently it returns null in this case